### PR TITLE
Feature/LJI-35-configure-redis

### PR DESCRIPTION
**Pull request by NormantasN**
Added redis configuration to docker-compose, created RedisConfig

### CODE REVIEW

**Review by anddevdev — APPROVED** [review]

**Comment by SavaLX** [comment]
Let's wait until the 31 PR will be merged

**Review by anddevdev — COMMENTED** [review]
Can we also add keycloak configuration?

**Comment by SavaLX** [comment]
@anddevdev  it would be the new task with separate ticket for the next sprint
@NormantasN Don't add keycloak to this PR

**Comment by NormantasN** [comment]
> @anddevdev it would be the new task with separate ticket for the next sprint @NormantasN Don't add keycloak to this PR

I think I have to because keycloak is already merged and to solve conflicts to merge I need to add keycloak

**Comment by SavaLX** [comment]
@NormantasN So yes, I meant you don't need to add docker configuration for keycloak, but add already existing config you should

File: docker-compose.yml [hunk 1]
```diff
@@ -12,21 +12,41 @@ services:
     profiles:
       - prod
 
+  redis:
```
Comment (SavaLX) [inline docker-compose.yml:15]: Let's give the names to all containers

File: docker-compose.yml [hunk 2]
```diff
@@ -12,21 +12,41 @@ services:
...
+      - dev
+      - prod
+
   insurance-app-dev:
     build: .
     environment:
       SPRING_PROFILES_ACTIVE: dev
+      REDIS_HOST: redis
+      REDIS_PORT: 6379
+      REDIS_PASSWORD: yourpassword
```
Comment (SavaLX) [inline docker-compose.yml:32]: Let's leave the password here, but FYI it's a bad and dangerous practice

File: src/main/java/com/coherentsolutions/pot/insurance/RedisConfig.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+
+@Configuration
+public class RedisConfig {
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/RedisConfig.java:11]: Take a look at #38

File: src/main/resources/application-prod.yml [hunk 4]
```diff
@@ -19,6 +19,11 @@ spring:
     web:
       pageable:
         default-page-size: 10
+    redis:
```
Comment (SavaLX) [inline src/main/resources/application-prod.yml:22]: As I see we have the same settings for both profiles, it may be better to extract it into one file.

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (NormantasN) [reply docker-compose.yml]: Done

File: src/main/java/com/coherentsolutions/pot/insurance/RedisConfig.java (hunk 3 above)
Comment (NormantasN) [reply src/main/java/com/coherentsolutions/pot/insurance/RedisConfig.java:25]: Added additional configuration

File: src/main/resources/application-prod.yml (hunk 4 above)
Comment (NormantasN) [reply src/main/resources/application-prod.yml]: Done

**Review by SavaLX — APPROVED** [review]

**Comment by SavaLX** [comment]
Your PR doesn't use the pipeline check, please add it and after successfully passing the build pipeline you will able to merge it

**Pull request merged**